### PR TITLE
Major bug fixes and minor code quality enhancements

### PR DESCRIPTION
**Pull request by Paulmski**
### Major Changes
- Improved handling of study group titles and special characters in titles.
- Added docstrings for `!help` command
- Added `!version` command to see the current version.
   * Releases will be tagged with a version for all future releases.

### Minor Changes
- Minor formatting fixes
- Refactored sheet_parser.py to be more reusable.

### TODO:
I have done my best to check for bugs and possible security vulnerabilities, but I am definitely imperfect. Those who I've requested a review from if you could give this release a thorough once over so that we don't have to do any more frantic bug fixes that would be much appreciated. Ideally, this pull request can be reviewed and merged by tomorrow morning.

### CODE REVIEW

**Comment by tuanyvan** [comment]
I will give it a look over tonight.

**Comment by Paulmski** [comment]
Feel free to make inline comments or start conversations about specific lines.

**Comment by Paulmski** [comment]
I'll be offline for the rest of the evening. I'll be up first thing in the morning to be available if stuff gets looked at later tonight. If the update is a go ahead it can be merged for the following days auto update.

File: src/__main__.py [hunk 1]
```diff
@@ -88,20 +104,9 @@ async def list(ctx, code=None):
         code = code.upper().replace('-', '').replace(' ','')
         assignments = sheets_parser.fetch_assignments(service, SPREADSHEET_ID, RANGE_NAME)
         final_assignments = []
-        current_course = ''
-        current_code = ''
         # Remove all courses that don't have a matching course code and aren't within 14 days.
         for assignment in assignments:
-            if assignment.code != '':
```
Comment (tuanyvan) [inline src/__main__.py:95]: Good call removing the massive number of if statements here. The logic is already handled by `events.py`.

File: src/__main__.py [hunk 2]
```diff
@@ -88,20 +104,9 @@ async def list(ctx, code=None):
...
-                current_course = assignment.course_name
-                current_code = assignment.code
-            if not -1 <= assignment.days_left <= 14:
-                continue
-            if assignment.code == "":
-                assignment.code = current_code
-            if code == 'ALL' or current_code == code:
-                assignment.course_name = current_course
-                assignment.code = current_code
+            if (code == 'ALL' or assignment.code == code) and -1 <= assignment.days_left <= 14:
```
Comment (tuanyvan) [inline src/__main__.py:109]: Good call removing the massive number of if statements here. The logic is already handled by `events.py`.

**Review by tuanyvan — APPROVED** [review]
All review changes requested are addressed in commit 31490f0 and PR #70.

The code changes made in this PR are critical to bot security and restores functionalities to commands like `!homework`. I have tested various edge cases, including command arguments not discussed before. The implementations in this PR prevented all attempts at malicious command usage and made logs of these attempts accordingly.

The method used to parse rows from the Google Sheets ensures that every row's `Assignment` object has some form of state in its `code` and `course_name` instance variables, which is much more stable and reusable than copy/pasting that behavior in other functions.

**Pull request merged**